documentation/upgrade.md

```diff
@@ -9,6 +14,37 @@ This version includes the following changes:
 * Updated to [operator-sdk v0.12.0](https://github.com/operator-framework/operator-sdk/releases/tag/v0.12.0)
 * Using Go modules instead of dep now 
 
+## Upgrade from 3.x.x to 4.x.x
+
+There is no direct upgrade path from previous versions to 4.0.0. This is due to an upgraded operator-sdk version and an update of the CRD definitions from v1beta1 to v1.
+
+To upgrade, we recommend the following steps:
+
+### Uninstall the previous version
+
+To uninstall the Grafana Operator, either remove the deployment with the name `grafana-operator`, or, if installed through OLM, follow the appropriate steps to remove the subscription:
+
+1) On OpenShift 4.x, you can uninstall the Operator via `Operators -> Installed Operators`
+2) On Kubernetes, use kubectl to identify the subscription:
+
+```shell
+$ kubectl get subscriptions -n<operator namespace>
+```
+
+Then delete the subscription:
+
+```shell
+$ kubectl delete subscription <subscription name> -n<operator namespace>
+```
+
+__NOTE__: uninstalling the Grafana Operator will not remove your Grafana instance or your dashboards.
+
+### Install 4.0.0
+
+Install the Grafana Operator v4.0.0 either using OLM or by [deployment](https://github.com/grafana-operator/grafana-operator/blob/v3.10.3/deploy/operator.yaml) or through OLM.
```

Comment:
I would ether link back to: https://github.com/grafana-operator/grafana-operator/blob/4d90d97b530d7626daf6eb02b9baeaf332c25e59/documentation/deploy_grafana.md#kustomize or write similar instructions in here.